test/common/chromium-cdp-driver.js

```diff
@@ -69,16 +69,20 @@ function clearExceptions() {
 }
 
 function stringifyConsoleArg(arg) {
-    if (arg.type === 'string')
-        return arg.value;
-    if (arg.type === 'object') {
-        const obj = {};
-        arg.preview.properties.forEach(prop => {
-            obj[prop.name] = prop.value.toString();
-        });
-        return JSON.stringify(obj);
+    try {
+        if (arg.type === 'string')
+            return arg.value;
+        if (arg.type === 'object') {
+            const obj = {};
+            arg.preview.properties.forEach(prop => {
+                obj[prop.name] = prop.value.toString();
+            });
+            return JSON.stringify(obj);
+        }
+        return JSON.stringify(arg);
+    } catch {
+        return "???";
```

Comment:
This is terrible to debug, though. Could this at least print the exception?

This catches two different cases: (1) `arg.preview.properties` not existing, and (2) JSON.stringify() failing on complex objects with e.g. recursive ref loops. The second commit avoids (1), but it would also be good to guard against it, with `if (arg.preview?.properties)` (and ignoring the `arg.type === 'object')`, or check for both).

Reply:
Yes, let's do that. 

(I think self-referencing objects can't happen, because we never get nested objects in the first place here. `x = {}; x.x = x` will be printed as `{"x":"Object"}`.)